### PR TITLE
Migrate contract package of iexec-common library

### PR DESCRIPTION
**Pull request by thewhitewizard**

### CODE REVIEW

File: src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskContributeArgs.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskContributeArgs.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskFinalizeArgs.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskFinalizeArgs.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskInitializeArgs.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskInitializeArgs.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskRevealArgs.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/adapter/args/TaskRevealArgs.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/main/java/com/iexec/commons/poco/chain/eip712/entity/Challenge.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/eip712/entity/Challenge.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/main/java/com/iexec/commons/poco/chain/eip712/entity/EIP712Challenge.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/eip712/entity/EIP712Challenge.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/main/java/com/iexec/commons/poco/chain/validation/EthereumNonZeroAddressValidator.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/validation/EthereumNonZeroAddressValidator.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/main/java/com/iexec/commons/poco/chain/validation/ValidNonZeroEthereumAddress.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/commons/poco/chain/validation/ValidNonZeroEthereumAddress.java:2]:
```suggestion
 * Copyright 2023 IEXEC BLOCKCHAIN TECH
```

File: src/test/java/com/iexec/commons/poco/utils/ChainUtilsTests.java
Comment (jbern0rd) [inline src/test/java/com/iexec/commons/poco/utils/ChainUtilsTests.java:1]: Should we move this class in the correct `chain` package immediately (or in a next PR) ?

File: build.gradle [hunk 9]
```diff
@@ -25,10 +26,53 @@ repositories {
...
     }
+    maven { url 'https://jitpack.io' }
 }
 
 dependencies {
+    implementation platform('org.springframework.boot:spring-boot-dependencies:2.6.14')
     api 'org.web3j:core:4.8.9'
+
+    //commons-io
+    implementation 'commons-io:commons-io:2.5'
```
Comment (jbern0rd) [inline build.gradle:37]: Does not seem to be necessary as we do not migrate `FileHelper`.

File: build.gradle [hunk 10]
```diff
@@ -25,10 +26,53 @@ repositories {
...
+    implementation 'org.apache.commons:commons-lang3:3.12.0'
+
+    // multiaddresses (IPFS)
+    implementation 'com.github.multiformats:java-multiaddr:v1.4.6'
+    
+    //jaxb required with Java 11
+    implementation 'javax.xml.bind:jaxb-api:2.3.1'
+    
+    // zip
+    implementation 'net.lingala.zip4j:zip4j:2.3.1'
```
[the remark this answers is not in the document]
Comment (thewhitewizard) [reply build.gradle:49]: We can remove zip4j but not jaxb which is used in BytesUtils (DatatypeConverter)

File: .gitignore [hunk 11]
```diff
@@ -1,14 +1,29 @@
...
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
 
-# # Work around https://youtrack.jetbrains.com/issue/IDEA-116898
-# gradle/wrapper/gradle-wrapper.properties
+/bin/
+.vscode/
```
Comment (jeremyjams) [inline .gitignore:29]: Very minor
```suggestion
.vscode/

```

File: CHANGELOG.md [hunk 12]
```diff
@@ -6,6 +6,7 @@ All notable changes to this project will be documented in this file.
 
 ### New Features
 - Init Gradle project. (#1 #2)
+- Migrate contract package of `iexec-common` library and the other classes used in the POCO protocol to this `iexec-commons-poco` library. (#3)
```
Comment (jeremyjams) [inline CHANGELOG.md:9]: Very minor
```suggestion
- Migrate from `iexec-common` library to this `iexec-commons-poco` library packages related to blockchain and iExec PoCo protocol. (#3) 
```
You can keep yours or another one no problem :+1:

File: src/main/java/com/iexec/commons/poco/chain/adapter/CommandStatus.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
+/*
```
Comment (jeremyjams) [inline src/main/java/com/iexec/commons/poco/chain/adapter/CommandStatus.java:1]: Shouldn't this package be in blockchain-adapter library?

File: src/main/java/com/iexec/commons/poco/security/Signature.java [hunk 14]
```diff
@@ -0,0 +1,64 @@
+/*
```
Comment (jeremyjams) [inline src/main/java/com/iexec/commons/poco/security/Signature.java:1]: Why do we have signature/, tee/ & utils/?

File: src/main/java/com/iexec/commons/poco/chain/adapter/CommandStatus.java (hunk 13 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/commons/poco/chain/adapter/CommandStatus.java:1]: it totally could be done like this.
What about doing it in a next step while deciding if endpoints related to `contribute` and `reveal` should be removed from blockchain-adapter as well ?

File: src/main/java/com/iexec/commons/poco/security/Signature.java (hunk 14 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/commons/poco/security/Signature.java:1]: `tee` package is required in TaskDescription
Part of `utils` package is required in `chain` package.
`signature` is required by `utils.SignatureUtils` used in `chain` package.

The intention here was to follow and include required packages and classes without heavily modifying things.
Tests and cleanups will come after this first code move (as iso as possible)
Some modifications will require more thorough analyses we don't have the time to perform today.

File: CHANGELOG.md (hunk 12 above)
Comment (thewhitewizard) [reply CHANGELOG.md]: ok,it's integrated

**Review by jbern0rd — APPROVED** [review]
Checks done:
- no major difference with https://github.com/iExecBlockchainComputing/iexec-common/commit/719bbbd4b512b805f7cbc38645de49ebcba7d0b7
- e2e test done with cleaned iexec-common, updated apps and some IT execution

**Pull request merged**